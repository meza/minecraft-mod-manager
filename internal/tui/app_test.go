package tui

import (
	"testing"

	tea "github.com/charmbracelet/bubbletea"
	"github.com/gkampitakis/go-snaps/snaps"
)

type dummyModel struct{}

func (d dummyModel) Init() tea.Cmd                           { return nil }
func (d dummyModel) Update(msg tea.Msg) (tea.Model, tea.Cmd) { return d, nil }
func (d dummyModel) View() string                            { return "content" }

func TestNewAppModel(t *testing.T) {
	t.Setenv("MMM_TEST", "true")
	m := NewAppModel([]Component{{Name: "Init", Model: dummyModel{}}})
	snaps.MatchSnapshot(t, m.View())
}
