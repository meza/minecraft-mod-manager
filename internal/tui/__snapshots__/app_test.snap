
[TestNewAppModel - 1]
──────────────────                                      
                     Select a command from the sidebar  
  ❯ Init                                                
                                                        
                                                        
──────────────────                                      
↑/↓ Navigate • ↵ Select • Q Quit • H Help               
---
